src/databricks/labs/ucx/contexts/application.py

```diff
@@ -571,6 +577,20 @@ def migration_recon(self) -> MigrationRecon:
     def administrator_locator(self) -> AdministratorLocator:
         return AdministratorLocator(self.workspace_client)
 
+    @cached_property
+    def ownership_factory(self) -> Callable[[Record], Ownership]:
+        # ensure registration of Ownerships
+        _ = [
+            self.directfs_access_ownership,
+            self.grant_ownership,
+            self.legacy_query_ownership,
+            self.table_migration_ownership,
+            self.table_ownership,
+            self.udf_ownership,
+            self.workspace_path_ownership,
+        ]
+        return Ownership.for_record
```

Comment:
As things stand during tests we re-create the application context for each test that needs one, and this is going to lead to the registry accumulating instances for each test that hits this property.

For this reason (and others) in the past we've preferred to avoid global singletons (which the class-hosted registry is), and move the cache to be owned and managed outside of it so that the lifecycle of the cache is tied to the application context.